openrtb_ext/imp_sharethrough.go

```diff
@@ -1,13 +1,18 @@
 package openrtb_ext
 
-type ExtImpSharethrough struct {
-	Pkey       string  `json:"pkey"`
-	Iframe     bool    `json:"iframe"`
-	IframeSize []int   `json:"iframeSize"`
-	BidFloor   float64 `json:"bidfloor"`
+type ExtData struct {
```

Comment:
Just checking, do you need `ExtData` or could you remove it and just have `PBAdSlot` in `ExtImpSharethrough` with an `omitempty`?

Reply:
@bsardo I am not sure. I am trying to get the GPID found in `imp[].ext.data`, similar to #1807.

In pbjs the method to implement was to extract the `ortb2Imp.ext.data.pbadslot` value. I am looking for the equivalent for prebid-server. As far as I can tell from checking PRs from other adapters they all implement it this way, but I was unable to find any official documentation that says exactly how this should be done. 

Can you help? I see that you approved the PR from `TheMediaGrid`, is there something that our implementation is missing?

Reply:
Sure I can help. I did some digging and I now believe your implementation is correct assuming the GPID, which I'm guessing stands for global placement ID, is considered ad-specific first party data.

The documentation here indicates that all PBS request ad-specific first party data belongs at `imp[].ext.data`:
https://docs.prebid.org/prebid-server/endpoints/openrtb2/pbs-endpoint-auction.html#prebid-server-ortb2-extension-summary

Reply:
@epechuzal, sorry I got confused above, hence the strikethrough. I still believe your implementation is correct but I wasn't clear why. Let me back up.

You're expecting `PbAdslot` and `AdServer` at `imp[].ext.data` in the PBS inbound request and you're putting `GPID` in the query string of the PBS outbound request to your server.
For inbound request data, if it is ad-specific first party data, it belongs at `imp[]ext.data` according to the referenced documentation above. It looks like `PbAdslot` and `AdServer` are indeed ad-specific and in the right place, similar to how The Media Grid does it in #1807.

As for `GPID`, you and The Media Grid are both setting it to `PbAdSlot` but in different places in your bid requests. You're setting it in each query string and they are setting it in each `imp.ext` which is fine since that is where your servers are expecting to find that data.

I hope that helps.

Reply:
@bsardo thank you for looking into it! let me know if there is anything that I should change for this PR to go in